astropy/cosmology/core.py

```diff
@@ -118,14 +119,13 @@ def __init_subclass__(cls):
         cls._init_signature = sig  # store (immutable) initialization signature
 
     def __new__(cls, *args, **kwargs):
-        # bundle initialization argument
+        self = super().__new__(cls)
```

Comment:
The change in `__new__` seems unrelated and makes things less clear, so I'd not do it.

Reply:
I'll move it to another PR. The MappingProxyType protection on `_init_arguments` is good for ensuring it is immutable.